### PR TITLE
feat(devtools): allow user to change frontmatter in multiple posts

### PR DESCRIPTION
**Pull request by smallp**
### Description
Enable user to change frontmatter in multiple posts.

### Linked Issues
#174 

### Copilot Descriptions

copilot:all

### CODE REVIEW

**Comment by WRXinYue** [comment]
My suggestion is to do this in a plugin or terminal, which can be developed and maintained independently without affecting existing development tools, rather than adding it to devtools. Adding too many non-core functions might make devtools bloated.

**Comment by YunYouJun** [comment]
Without adding additional dependencies, the added code is manageable. Maybe we can design a plug-in system for devtools in the future and get it up and running before we do that.

**Comment by YunYouJun** [comment]
cc @WRXinYue  @qtqz @guowei-gong 

Are anyone interested in verifying and refining this feature?

File: packages/devtools/src/node/api/index.ts
Comment (YunYouJun) [inline packages/devtools/src/node/api/index.ts]: All `userroot` should be `userRoot`

File: packages/devtools/src/node/index.ts
Comment (YunYouJun) [inline packages/devtools/src/node/index.ts]: Why redefine?

**Review by YunYouJun — CHANGES_REQUESTED** [review]

File: packages/devtools/src/node/api/index.ts
Comment (YunYouJun) [inline packages/devtools/src/node/api/index.ts]: `pageData.path` is a full path. You can not use it like this.

File: packages/valaxy/node/server.ts
Comment (YunYouJun) [inline packages/valaxy/node/server.ts]: useRoot can got from `options`. You should not pass it.

**Review by YunYouJun — COMMENTED** [review]
image

This pr contains the wrong type build (js-yaml) so that it cannot pass CI. For new features, and there is no style, no comments.
Before waiting for a merge, you should make sure it passes all CIS and resolves conflicts.

File: packages/devtools/src/node/api/index.ts
Comment (YunYouJun) [inline packages/devtools/src/node/api/index.ts]: The meaning of pagedata here does not correspond to its meaning.

**Comment by YunYouJun** [comment]
Just merge it for later work.

File: packages/valaxy/node/server.ts
Comment (smallp) [reply packages/valaxy/node/server.ts]: That's because I dont know how to access userRoot in options.

Comment (YunYouJun) [reply packages/valaxy/node/server.ts]: Get it from `options.userRoot` directly.

**Pull request merged**
